assets/app/views/_templateopt.html

```diff
@@ -22,11 +22,19 @@
           type="text"
           placeholder="{{ parameter | parameterPlaceholder }}"
           ng-model="parameter.value"
-          ng-required="parameter.required && !parameter.generate">
+          ng-required="parameter.required && !parameter.generate"
+          ng-trim="false">
     </span>
     <div class="help-block" ng-if="parameter.description">{{parameter.description}}</div>
     <div ng-show="paramForm[paramID].$error.required && paramForm[paramID].$touched" class="has-error">
-      <div class="help-block">{{parameter.name}} is required.</div>
+      <div class="help-block">{{parameter.displayName || parameter.name}} is required.</div>
+    </div>
+    <div ng-show="isOnlyWhitespace(parameter.value)" class="has-warning">
+      <span class="help-block">
+        The current value only has spaces.
```

Comment:
```
The current value is "{{parameter.value}}", which is not empty.
<span ng-if="parameter.generate">This will prevent a value from being generated.</span>
If this isn't what you want, <a ng-click="parameter.value=''">clear value</a>
```